### PR TITLE
meta: update eslint config

### PR DESCRIPTION
**Pull request by ephys**
## Pull Request Checklist

- [x] Does the name of your PR follow [our conventions](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#6-commit-your-modifications)?

## Description Of Change

I've published a new version of https://github.com/ephys/eslint-config to update all plugins & add new relevant rules, this is the result.

The main change is that `@ts-expect-error` requires a comment, and that comment must follow the same format as eslint's (for consistency)

### CODE REVIEW

File: test/types/where.ts [hunk 1]
```diff
@@ -144,27 +144,27 @@ MyModel.findAll({
...
   },
 });
 
-// @ts-expect-error - no attribute
+// @ts-expect-error -- no attribute
 MyModel.findAll({
   where: [1, 2],
 });
 
 // TODO [2022-05-26]: TS < 4.4 does not detect an error here. Uncomment test once we remove support for TS 4.3
```
Comment (WikiRik) [inline test/types/where.ts:159]: Not directly related to this PR, but I believe these tests are uncommented now and this TODO can be removed, also holds for a few other instances in this file

File: test/integration/sequelize.test.js [hunk 2]
```diff
@@ -462,15 +462,31 @@ describe(Support.getTestDialectTeaser('Sequelize'), () => {
...
-          } else if (dialect === 'mssql') {
-            expect(error.message).to.include('Login failed for user \'bar\'.');
-          } else if (dialect === 'db2') {
-            expect(error.message).to.include('A communication error has been detected');
-          } else if (dialect === 'ibmi') {
-            expect(error.message).to.equal('[odbc] Error connecting to the database');
-            expect(error.original.odbcErrors[0].message).to.include('Data source name not found and no default driver specified');
           } else {
-            expect(error.message.toString()).to.match(/.*Access denied.*/);
+            switch (dialect) {
```
Comment (WikiRik) [inline test/integration/sequelize.test.js:466]: There is no `postgres-native` for `dialect` so we can use the switch for all dialects. See https://github.com/sequelize/sequelize/blob/344c404eac7393fd5d460cb65eca835c764763d2/test/support.ts#L253-L265

File: test/integration/configuration.test.js [hunk 3]
```diff
@@ -60,15 +60,29 @@ describe(Support.getTestDialectTeaser('Configuration'), () => {
       }
 
       const seq = new Sequelize(config[dialect].database, config[dialect].username, 'fakepass123', { logging: false, host: config[dialect].host, port: 1, dialect });
-      if (dialect === 'sqlite') {
+      switch (dialect) {
+        case 'sqlite': {
         // SQLite doesn't require authentication and `select 1 as hello` is a valid query, so this should be fulfilled not rejected for it.
```
Comment (WikiRik) [inline test/integration/configuration.test.js:65]: I think we should inline this a little bit

Comment (ephys) [reply test/integration/configuration.test.js:65]: I actually didn't mean to convert to switch, it's a rule we disabled in our local config, that I disabled in the base config, but it I auto-fixed before updating the base config :/

I'll revert the switch changes

File: test/integration/sequelize.test.js (hunk 2 above)
Comment (WikiRik) [reply test/integration/sequelize.test.js:466]: Just adding two extra spaces at the start so the indentation of the exepct is the same as the comment

File: test/integration/configuration.test.js (hunk 3 above)
Comment (WikiRik) [reply test/integration/configuration.test.js:65]: Just adding two extra spaces at the start so the indentation of the exepct is the same as the comment

File: test/types/associations.ts [hunk 4]
```diff
@@ -30,7 +30,7 @@ Car.belongsTo(Person, { foreignKey: 'doesnotexist' });
 Car.belongsTo(Person, { foreignKey: { name: 'doesnotexist' } });
 
 Car.belongsTo(Person, { targetKey: 'id' });
-// @ts-expect-error
+// @ts-expect-error -- this should error, if this doesn't error, findOne has a bug!
```
[the remark this answers is not in the document]
Comment (ephys) [reply test/types/associations.ts:33]: Done

File: test/types/where.ts (hunk 1 above)
Comment (ephys) [reply test/types/where.ts:159]: Done

File: test/integration/sequelize.test.js (hunk 2 above)
Comment (ephys) [reply test/integration/sequelize.test.js:466]: Done

**Review by WikiRik — APPROVED** [review]
Looks good! I'll approve again once the conflicts have been resolved

**Comment by ephys** [comment]
Here you go :)

**Review by WikiRik — APPROVED** [review]

**Pull request merged**